vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideExternalLinkManager.java

```diff
@@ -0,0 +1,79 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.SlideExternalLinkRepository;
+import edu.asu.diging.vspace.core.model.IExternalLinkSlide;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.ExternalLinkSlide;
+import edu.asu.diging.vspace.core.services.ISlideExternalLinkManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+
+@Transactional
+@Service
+public class SlideExternalLinkManager
+        implements ISlideExternalLinkManager {
+
+    @Autowired
+    private ISlideManager slideManager;
+
+    @Autowired
+    private SlideExternalLinkRepository externalLinkRepo;
+
+    @Override
+    public IExternalLinkSlide getLink(String externalLinkID) {
+        Optional<ExternalLinkSlide> externalLink = externalLinkRepo.findById(externalLinkID);
+        if (externalLink.isPresent()) {
+            return externalLink.get();
+        }
+        return null;
+    }
+    
+    @Override
+    public void addToLinkList(ISlide slide, IExternalLinkSlide link) {
+        slide.getExternalLinks().add(link);
+    }
+
+    @Override
+    public void removeFromLinkList(ISlide slide, IExternalLinkSlide link) {
```

Comment:
does this really need a method? when you want to remove a link, you can just do it directly without calling this extra method.